84-sarcastic-fringehead/technical_specification.md

```diff
@@ -0,0 +1,478 @@
+As for the work to be completed within this epic, the services affected include the File Ingest Service (FIS), Internal File Registry Service (IFRS), the Well-Known Value Service (WKVS), the Upload Controller Service (UCS), the ghga-event-schemas library, and a new service called the Data Hub File Service (DHFS). Additionally, if it is discovered during implementation that further changes need to be made to other services *beyond what is described in this epic*, then tickets will be added ad-hoc and associated with this epic.
+
+In Lynx Boreal, the UCS was rewritten, the Upload Orchestration Service (UOS) was implemented for the first time, the Claims Repository Service (CRS) was updated to manage permissions for Research Data Upload Boxes, and the Work Package Service (WPS) was updated to manage upload-type work packages. Taken together, these changes create the operational framework for remote file upload, but only to the point of initial ingest. In order to fully realize our file upload concept, we still need to decrypt the uploaded file, verify the integrity via checksum comparison, re-encrypt the file with a new file secret (securely stored in the Encryption Key Store Service, or EKSS), and move the file to a permanent storage bucket registered with the IFRS in what we call "archival".
+
+
+### Included/Required:
+All work described in the Additional Implementation Details section below is required.
+
+### Optional:
+Add email notifications for important events related to archival. This could include, for example, a notification conveying that all files in a Research Data Upload Box have been successfully archived, or that there was a problem with file XYZ during interrogation. To prevent scope creep, this should *probably* be done in another epic, but we should keep that potential requirement in mind during development.
+
+Add audit events -- can also be performed in a separate epic.
+
+### Not included:
+Data Portal updates or any upcoming metadata-related services. This is purely for file upload.
+
+## User Journeys (optional)
+
+All user journeys are already detailed in Lynx Boreal. The operations added in this epic will occur automatically without further action required on the part of either the user or GHGA personnel. 
+
+## API Definitions:
+
+### RESTful/Synchronous:
+
+FIS:
+> See the [FIS HTTP API](#fis-http-api) section for an in-depth explanation
+
+- `POST /secrets`
+  - Authorization requires a token signed with Data Hub-specific private key
+  - Request body must contain an encrypted file secret
+  - Returns `201 CREATED`
+- `GET /uploads/{storage_alias}/inbox`
+  - Authorization requires a token signed with Data Hub-specific private key
+  - Returns `200 OK` and a list of `FileUploads` for files awaiting interrogation
+  - Files are grouped by storage alias
+- `GET /reports/{file_id}`
```

Comment:
In a RESTful API, I would expect this to be called "inspection-results" instead of "reports". Consistent naming makes it a bit easier to understand this already pretty complex process.